optuna/storages/_rdb/storage.py

```diff
@@ -143,7 +143,8 @@ def objective(trial):
         :exc:`ValueError`:
             If the given `heartbeat_interval` or `grace_period` is not a positive integer.
         :exc:`RuntimeError`:
-            If the a process that was failed by heartbeat but was actually running.
+            When a process tries to finish a trial that has already
+            been set to FAILED by heartbeat.
```

Comment:
A nit, but how about adding a link to `TrialState.FAIL`?
```suggestion
            been set to :class:`~optuna.trial.TrialState.FAIL` by heartbeat.
```